astropy/time/formats.py

```diff
@@ -664,6 +664,77 @@ def to_value(self, timezone=None, parent=None):
     value = property(to_value)
 
 
+class TimeYMDHMS(TimeUnique):
+    """
+    ymdhms: A Time format to represent Time as year, month, day, hour,
+    minute, second (thus the name ymdhms)
+    Example::
+
+      >>> from astropy.time import Time
+      >>> t = Time({'year': [2015, 2016], 'minute': [1, 2]}, scale='utc')
+      >>> t.iso
+      array(['2015-01-01 00:01:00.000', '2016-01-01 00:02:00.000'], dtype='<U23')
+      >>> t.ymdhms['year']
+      array([2015, 2016], dtype=int32)
+    """
+    name = 'ymdhms'
+
+    def _check_val_type(self, val1, val2):
+        # Note: don't care about val2 for this class
+        if not all(isinstance(val, dict) or
+                   (isinstance(val, np.record) and val.dtype.kind == 'V')
+                   for val in val1.flat):
+            raise TypeError('Input values for {0} class must be '
+                            'dict or numpy recarray or numpy structured'
+                            ' array objects'.format(self.name))
+        return val1, None
+
+    def set_jds(self, val1, val2):
+        if val1.dtype.kind == 'V':
+            times_dict = dict([(dtype, val1[dtype])
+                               for dtype in val1.dtype.names])
+        else:
+            times_dict = val1.item()
+
+        jd1, jd2 = erfa.dtf2d(self.scale.upper().encode('ascii'),
+                              times_dict['year'],
+                              times_dict.get('month', 1),
+                              times_dict.get('day', 1),
+                              times_dict.get('hour', 0),
+                              times_dict.get('minute', 0),
+                              times_dict.get('second', 0))
+        self.jd1, self.jd2 = day_frac(jd1, jd2)
+
+    @property
+    def value(self):
+        scale = self.scale.upper().encode('ascii')
+        iys, ims, ids, ihmsfs = erfa.d2dtf(scale, self.precision,
+                                           self.jd1, self.jd2_filled)
+        ihrs = ihmsfs['h']
+        imins = ihmsfs['m']
+        isecs = ihmsfs['s']
+        ifracs = ihmsfs['f']
+
+        shape = 1 if not self.jd1.shape else self.jd1.shape
+        out = np.empty(shape, dtype=[
```

Comment:
I think this should be `out = np.empty(self.jd1.shape, dtype=[ ... ])`.  For a scalar this will automatically give a scalar output.  Shape guru @mhvk?  l guess we have an API question of whether the scalar case is consistent with the array case in terms of returning typed numpy objects, or is different and returns a pure Python dict of float/int.  Somehow I prefer the consistent output, but in terms of performance and interface the Python dict might be better.
```
In [18]: out  # structured array with shape=()
Out[18]: 
array((2018, 0, 0, 0, 0, 0.),
      dtype=[('year', '<i4'), ('month', '<i4'), ('day', '<i4'), ('hour', '<i4'), ('minute', '<i4'), ('second', '<f8')])

In [19]: out['year']
Out[19]: array(2018, dtype=int32)
```

Reply:
@mhvk suggested me the same. When I used that I got a 0d array. That's why I changed. I will switch in the next commit.

Reply:
> Somehow I prefer the consistent output, but in terms of performance and interface the Python dict might be better.

I vote for consistent output too.

Reply:
the scalar case of a structured array is a `void` and for `recarray` it is a `record`. I agree that we should use those for consistency.

Reply:
p.s. The easiest way to get the scalar is by creating the array version and then doing `out = out[()] if not out.shape else out`

Reply:
I could not do this. When I set the shape as `()`, `Time.ymdhms` is a plain array (a tuple, I think) and not a rec array anymore.

Any hints how to tackle this?

Reply:
@vn-ki - that was a tricky one: you also need to change `_shaped_like_input` to read `out = value[()] if value.dtype.fields else value.item()`